controllers/apps/apimanager_status_reconciler.go

```diff
@@ -42,9 +42,14 @@ func (s *APIManagerStatusReconciler) Reconcile() (reconcile.Result, error) {
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("failed to calculate status: %w", err)
 	}
-
-	equalStatus := s.apimanagerResource.Status.Equals(newStatus, s.logger)
-	s.logger.V(1).Info("Status", "status is different", !equalStatus)
+	var equalStatus bool
```

Comment:
although I can see a bit of room for improvement around this logic I'm struggling to understand why are we doing all these checks below? 

Why are these additional 2 checks required?

Reply:
Orignally flow was
```go
equalStatus = s.apimanagerResource.Status.Equals(newStatus, s.logger)
s.logger.V(1).Info("Status", "status is different", !equalStatus)
```
was reverting the status update on every reconcile 

The first check is to check weather the current state of the apimanger is avaiable if so don't update again. 

The second check is to check if the current state of the status is available i.e. there is an edge case where if you delete a deployment  and it is restored you will have a case where newStatus is available but s.apimangerResouces is not.

Reply:
hmm, I think we might have a different bug somewhere then. I've reverted your changes locally and it works fine. Can you tell me how to exactly reproduce the issue with old configuration?

Reply:
Enable hpa on backend with out logical db to get the status update you will notice that the message will get removed every reconcile. 

if you enable the first check on newStatus the status message doesn't get removed.

delete apicast production you will see that the status never corrects its self

set the second check for s.apimanagerResources, and you will see it correct it's self

Reply:
I think we need to find a better way of adding an info status to the CRs in general or at least agree on that whether we want to have separate calls to update status of CR within the code. IMO any status updates should be done as part of the status reconcilers. One of the examples on how could we achieve this is in this test PR: https://github.com/3scale/3scale-operator/pull/905

Not a huge fan of it due to the fact that we are adding a lot of extra code just to cover some use cases. There's probably a better solution...

Just looking at how the backends reconciler is adding the status seems wrong to me since we are always trying to replace the condition[0]. Maybe we can sync after xmas break and discuss how to achieve info message status reporting.

Reply:
removed this code